File: backend/app/schemas/conversation.py
from datetime import datetime

from pydantic import BaseModel


class ConversationCreate(BaseModel):
    agent_id: str | None = None
    agent_ids: list[str] | None = None  # for roundtable
    collaboration_mode: str | None = None  # "roundtable"
    model: str | None = None  # required if agent_id is None and not roundtable
    title: str | None = None


class MessageSend(BaseModel):
    content: str


class MessageResponse(BaseModel):
    id: str
    role: str
    content: str
    agent_id: str | None
    agent_name: str | None = None
    model_used: str | None
    created_at: datetime


class ConversationResponse(BaseModel):
    id: str
    title: str | None
    agent_id: str | None
    agent_ids: list[str] = []
    model: str | None
    is_collaboration: bool = False
    collaboration_mode: str | None = None
    message_count: int
    created_at: datetime
    updated_at: datetime


class ConversationDetailResponse(ConversationResponse):
    messages: list[MessageResponse]

File: frontend/src/types/conversation.ts
export interface Message {
  id: string;
  role: "user" | "assistant" | "system";
  content: string;
  agent_id: string | null;
  agent_name: string | null;
  model_used: string | null;
  created_at: string;
}

export interface Conversation {
  id: string;
  title: string | null;
  agent_id: string | null;
  agent_ids: string[];
  model: string | null;
  is_collaboration: boolean;
  collaboration_mode: string | null;
  message_count: number;
  created_at: string;
  updated_at: string;
}

export interface ConversationDetail extends Conversation {
  messages: Message[];
}
